vdr/api/v1/api.go

```diff
@@ -128,23 +149,28 @@ func (a Wrapper) UpdateDID(ctx echo.Context, targetDID string) error {
 		return ctx.String(http.StatusBadRequest, fmt.Sprintf("given hash is not valid: %s", err.Error()))
 	}
 
-	err = a.VDR.Update(*d, h, req.Document, nil)
+	err = a.VDR.Update(*id, h, req.Document, nil)
 	if err != nil {
-		return handleError(ctx, err, "could not update document: %s")
+		return handleError(ctx, err, updateErrTemplate)
 	}
 	return ctx.JSON(http.StatusOK, req.Document)
 }
 
 // DeactivateDID deactivates a DID Document given a DID.
 // It returns a 200 and an empty body if the deactivation was successful.
-func (a *Wrapper) DeactivateDID(ctx echo.Context, targetDID string) error {
-	id, err := did.ParseDID(targetDID)
+func (a *Wrapper) DeactivateDID(ctx echo.Context, didStr string) error {
+	id, err := did.ParseDID(didStr)
 	if err != nil {
 		return ctx.String(http.StatusBadRequest, fmt.Sprintf("given DID could not be parsed: %s", err.Error()))
 	}
+	// Check if org is managed by this node
+	err = a.OwnershipChecker.OwnedByThisNode(*id)
```

Comment:
this validation step should be performed by VDR service implementation, because now you might have to call it from multiple locations before deactivating/updating.

Reply:
Yes, that is the idea. I need to call it from API's so the application can easily check if an operation is permitted. To make everything dependant on VDR seems wrong, since that is manly for working with documents from the data registry: Reesolve, Create and Update.
This check is not only needed by VDR but also by auth/notary. Therfore it is put in a seperate function.

Reply:
If Auth/Notary needs `OwnershipChecker` which in turns needs `VDR` (through `KeyResolver`), Auth/Notary still transitively depend on VDR. So I wouldn't let it depend on VDR, but on the `OwnershipChecker` (to be renamed?) interface which is implemented by VDR?

Reply:
And I still think validating business rules in the API layer which aren't validated in the service layer is bad practice. So whether `OwnershipChecker` is implemented by `VDR` or not, `VDR` itself should call `OwnershipChecker`. API layer is a presentation layer, only responsible for translating between HTTP/REST and Go services.

Reply:
The check is input validation, the business logic doesn't care if this check succeeds or not. It prevents a more annoying error in the next API call. Placing it in the business logic creates more spaghetti code than placing it here.